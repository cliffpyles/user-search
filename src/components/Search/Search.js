import React from 'react'
import TextField from '../TextField'
import Button from '../Button'

import './Search.css'

export default function Search({ id, label, name, onChange, onSearch, value }) {
  return (
    <div className="Search">
      <TextField
        id={id}
        label={label}
        name={name}
        onChange={onChange}
        type="search"
        value={value}
      />
      <Button onClick={onSearch}>search</Button>
    </div>
  )
}
