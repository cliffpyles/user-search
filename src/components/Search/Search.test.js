import React from 'react'
import Renderer from 'react-test-renderer'
import Search from './'

test('can render a Search', () => {
  const component = Renderer.create(
    <Search
      id="example"
      name="example"
      label="example"
      onChange={() => {
        console.log('changing')
      }}
      onSearch={() => {
        console.log('searching')
      }}
      value="search for something"
    />
  )

  expect(component.toJSON()).toMatchSnapshot()
})
