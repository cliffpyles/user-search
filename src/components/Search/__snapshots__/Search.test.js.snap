// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a Search 1`] = `
<div
  className="Search"
>
  <div
    className="TextField"
  >
    <label
      htmlFor="example"
    >
      example
    </label>
    <input
      id="example"
      name="example"
      onChange={[Function]}
      type="search"
      value="search for something"
    />
  </div>
  <button
    className="Button"
    type="button"
  >
    search
  </button>
</div>
`;
